src/opnsense/www/js/widgets/Certificates.js

```diff
@@ -0,0 +1,177 @@
+ * AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * AUTHOR BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY,
+ * OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+export default class Certificates extends BaseTableWidget {
+
+    constructor(config) {
+        super(config);
+        this.configurable = true;
+        this.configChanged = false;
+    }
+
+    getGridOptions() {
+        return {
+            sizeToContent: 650
+        };
+    }
+
+    getMarkup() {
+        const $container = $('<div></div>');
+        const $certificateTable = this.createTable('certificateTable', {
+            headerPosition: 'none'
+        });
+
+        $container.append($certificateTable);
+        return $container;
+    }
+
+    async onWidgetTick() {
+        const cas = (await this.ajaxCall('/api/trust/ca/search')).rows || [];
+        const certs = (await this.ajaxCall('/api/trust/cert/search')).rows || [];
+
+        if (cas.length === 0 && certs.length === 0) {
+            this.displayError(`${this.translations.noitems}`);
+            return;
+        }
+
+        this.clearError();
+        await this.processCertificates(cas, certs);
+    }
+
+    displayError(message) {
+        const $error = $(`<div class="error-message">${message}</div>`);
+        $('#certificateTable').empty().append($error);
+    }
+
+    clearError() {
+        $('#certificateTable .error-message').remove();
+    }
+
+    processItems(items, type, hiddenItems, rows) {
+        items.forEach(item => {
+            if (!hiddenItems.includes(item.descr)) {
+                const validTo = new Date(parseInt(item.valid_to) * 1000);
+                const now = new Date();
+                const remainingDays = Math.max(0, Math.floor((validTo - now) / (1000 * 60 * 60 * 24)));
+
+                const colorClass = remainingDays === 0
+                    ? 'text-danger'
+                    : remainingDays < 14
+                        ? 'text-warning'
+                        : 'text-success';
+
+                const statusText = remainingDays === 0 ? this.translations.expired : this.translations.valid;
+
+                const row = `
+                    <div>
+                        <i class="fa fa-lock ${colorClass} certificate-tooltip" style="cursor: pointer;"
+                            data-tooltip="${type}-${item.descr}" title="${statusText}">
+                        </i>
+                        &nbsp;
+                        <span><b>${item.descr}${type === 'ca' ? ' (CA)' : ''}</b></span>
```

Comment:
Not sure this is needed, in all sane cases the subject should be clear

Reply:
(and a CA could also be a self-signed certificate)

Reply:
Yeah you are right.